### PR TITLE
FIX-#6687: Explicitly add users to CODEOWNERS

### LINKED ISSUE #6687: Explicitly add individual team members to CODEOWNERS file
We are planning to convert existing organization members to outside collaborators, which means they will lose their team. To avoid anyone losing existing permissions (particularly the modin core team) we need to explicitly add members to that file.

### PR DESCRIPTION
**Pull request by devin-petersohn**
Resolves #6687

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6687  
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # These owners will be the default owners for everything in
 # the repo unless a later match takes precedence,
-*    @modin-project/modin-core
+*    @modin-project/modin-core @devin-petersohn @mvashishtha @RehanSD @YarShev @vnlitvinov @anmyachev @dchigarev
 
 # These owners will review everything in the HDK engine component
 # of Modin.
-/modin/experimental/core/storage_formats/hdk/**    @modin-project/modin-hdk
-/modin/experimental/core/execution/native/implementations/hdk_on_native/**    @modin-project/modin-hdk
+/modin/experimental/core/storage_formats/hdk/**    @modin-project/modin-hdk @aregm @gshimansky @fexolm @ienkovich @Garra1980 @YarShev @vnlitvinov @anmyachev @dchigarev
```
Comment (Garra1980) [inline CODEOWNERS:7]: we can remove @fexolm here and below

Comment (Garra1980) [reply CODEOWNERS]: oh, and please add @AndreyPavlenko for hdk and hdk_on_native

**Review by RehanSD — DISMISSED** [review]
LGTM

**Review by Garra1980 — APPROVED** [review]

**Review by RehanSD — APPROVED** [review]
LGTM!

**Pull request merged**